### ISSUE TITLE
neovim displays wide code-areas of configure.ac as comments, not as code, partially fixed by Ctrl-L

### ISSUE THREAD

**Issue by toogley** [neovim/neovim#6850]
- `nvim --version`: NVIM v0.2.1-279-g018383096 (current master, installed with sudo make install)
- Vim (version: ) behaves differently? **Yes** (VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Apr 26 2017 07:50:37) Included patches: 1-586)
- Operating system/version: newest arch linux.
- Terminal name/version: gnome terminal 3.24.2-1
- `$TERM`: xterm-256color

### Actual behaviour

![2017-06-05-230006_swaygrab](https://cloud.githubusercontent.com/assets/9417950/26802953/c6aa911e-4a42-11e7-96e7-3c09bbc04b05.png)

On the right side is neovim, on the left side is vim. Its both the [exact same file and its exact same line](https://github.com/neomutt/neomutt/blob/b9bbe569e0eabfa27f9bc49f6f656f0f7b742992/configure.ac#L481-L520).

Both neovim and vim do not display the syntax highlighting for `configure.ac` correctly. Vim displays the syntax highlighting incorrectly only until line 514 ([which is this area](https://github.com/neomutt/neomutt/blob/b9bbe569e0eabfa27f9bc49f6f656f0f7b742992/configure.ac#L481-L514), but neovim displays the syntax highlighting incorrectly until much longer. See the screenshot for what i mean.

The first line displayed in both programs is correct as expected. I left it there as an orientation point.

I'm not sure about the last point, because i can **kind of** fix the syntax highlighting, **at least in some parts** of the incorrectly displayed buffer with pressing `Ctrl-L`. Maybe i can make a gif of the observed behavior.

### Expected behaviour

![2017-06-05-232641_swaygrab](https://cloud.githubusercontent.com/assets/9417950/26803941/78ffe3a2-4a46-11e7-830d-a706c88d2695.png)

Basically, i expect neovim to look similar to the above picture in the area described above.

### Steps to reproduce using `nvim -u NORC`

```
wget https://raw.githubusercontent.com/neomutt/neomutt/b9bbe569e0eabfa27f9bc49f6f656f0f7b742992/configure.ac ~/configure.ac
nvim -u NORC ~/configure.ac +481

```

**Comment by justinmk** [comment]
:syntax sync fromstart

**Comment by toogley** [comment]
@justinmk That is not a solution for my problem, although i have set this option.

**Comment by justinmk** [comment]
It's a runtime issue. When pointing VIMRUNTIME to that of Vim HEAD, nvim's behavior is identical to Vim HEAD add8dce38de65a0c64e8f54d6bdcadb45a8de2cf

**Comment by toogley** [comment]
So is there a solution for this runtime problem? or is this a bug? (if its a bug, should this reopened?)

**Comment by justinmk** [comment]
We merge Vim's runtime updates regularly.
